core/shared/src/main/scala/fs2/concurrent/Signal.scala

```diff
@@ -341,7 +349,38 @@ object SignallingRef {
       ref: SignallingRef[F, A]
   )(get: A => B, set: A => B => A)(implicit F: Functor[F]): SignallingRef[F, B] =
     new LensSignallingRef(ref)(get, set)
-
+  final private class TransformedSignallingRef[F[_], G[_], A](
+      underlying: SignallingRef[F, A],
+      trans: FunctionK[F, G]
+  )(implicit G: Functor[G])
+      extends SignallingRef[G, A] {
+
+    // --- Ref methods: these are lifted using trans, just like in TransformedRef2
+    override def get: G[A] = trans(underlying.get)
+    override def set(a: A): G[Unit] = trans(underlying.set(a))
+    override def getAndSet(a: A): G[A] = trans(underlying.getAndSet(a))
+    override def tryUpdate(f: A => A): G[Boolean] = trans(underlying.tryUpdate(f))
+    override def tryModify[B](f: A => (A, B)): G[Option[B]] = trans(underlying.tryModify(f))
+    override def update(f: A => A): G[Unit] = trans(underlying.update(f))
+    override def modify[B](f: A => (A, B)): G[B] = trans(underlying.modify(f))
+    override def tryModifyState[B](state: State[A, B]): G[Option[B]] =
+      trans(underlying.tryModifyState(state))
+    override def modifyState[B](state: State[A, B]): G[B] = trans(underlying.modifyState(state))
+    override def access: G[(A, A => G[Boolean])] =
+      G.compose[(A, *)].compose[A => *].map(trans(underlying.access))(trans(_))
+
+    // --- Signal-specific methods
+    override def discrete: Stream[G, A] = underlying.discrete.translate(trans)
+    override def continuous: Stream[G, A] = underlying.continuous.translate(trans)
+    override def changes(implicit eqA: Eq[A]): Signal[G, A] =
+      new Signal[G, A] {
+        def discrete = TransformedSignallingRef.this.discrete.changes
+        def continuous = TransformedSignallingRef.this.continuous
+        def get = TransformedSignallingRef.this.get
+      }
+    override def waitUntil(p: A => Boolean)(implicit G: Concurrent[G]): G[Unit] =
+      TransformedSignallingRef.this.discrete.forall(a => !p(a)).compile.drain
```

Comment:
Thanks for adding these. Unfortunately, this is just duplicating the default implementations. We already inherit these default implementations, so we don't need to duplicate it.

Instead, we should override these with implementations that delegate to `underlying`. The reason we want to do this, is because `underlying` may have overridden the default implementation with an optimized implementation. We want to prefer to use the optimized implementation whenever possible.